proposals/init.md

```diff
@@ -0,0 +1,490 @@
+}
+
+class Derived : Base
+{
+    Derived()
+    {
+        // Not allowed with get only properties but allowed with init
+        Value = true;
+    }
+}
+
+class Consumption
+{
+    void Example()
+    {
+        var d = new Derived() { Value = true; };
+    }
+}
+
+```
+
+At the point a `init` accessor is invoked the instance is known to be 
+in the open construction phase. Hence an `init` accessor is allowed to take 
+the following actions:
+1. Call other `init` accessors available through `this`
+1. Assign `readonly` fields declared on the same type
+
+```cs
+class Complex
+{
+    readonly int Field1;
+    int Field2;
+    int Prop1 { get; init ; }
+    int Prop2
+    {
+        get => 42;
+        set
+        {
+            Field1 = 13; // okay
+            Field2 = 13; // okay
+            Prop1 = 13; // okay
+        }
+    }
+}
+```
+
+The ability to assign fields from an `init` accessor is limited to fields
+declared on the same type as the accessor. It cannot be used to assign fields
+in a base type. This rule ensures that type authors remain in control over the
+mutability behavior of their type. Developers who do not wish to utilize 
+`init` cannot be impacted from other types choosing to do so:
+
+```cs
+class Base
+{
+    internal readonly int Field;
+    internal int Property
+    {
+        get => Field;
+        init => Field = value; // Okay
+    }
+}
+
+class Derived : Base
+{
+    internal readonly int DerivedField;
+    internal int DerivedProperty
+    {
+        get => DerivedField;
+        init
+        {
+            DerivedField = 42; // Okay
+            Field = 13; // Error Field is readonly
+        }
+    }
+
+    public Derived()
+    {
+        Property = 42; // Okay 
+        Field = 13; // Error Field is readonly
+    }
+}
+```
+
+When `init` is used in a virtual property then all the overrides must also
+be marked as `init`. Likewise it is not possible to override a simple 
+`set` with `init`.
+
+An `interface` declaration can also particpate in `init` style initalization 
+via the following pattern:
+
+```cs
+interface IPerson
+{
+    string Name { get; init; }
+}
+
+class Init
+{
+    void M<T>() where T : IPerson, new()
+    {
+        var local = new T()
+        {
+            Name = "Jared"
+        };
+        local.Name = "Jraed"; // Error
+    }
+}
+```
+
+Restrictions of this feature:
+- The `init` accessor can only be used instance properties
+- A property cannot contain both an `init` and `set` accessor
+- All overrides of a property must have `init` if the base had `init`. This rule
+also applies to interface implementation.
+
+### Metadata encoding 
+Property `init` accessors will be emitted as a standard `set` accessor with
+the return type marked with a modreq of `IsInitOnly`. This is a new type
+which will have the following definition:
+
+```cs
+namespace System.Runtime.CompilerServices
+{
+    [AttributeUsage(AttributeTargets.All)]
+    public sealed class InitOnlyAttribute : Attribute
```

Comment:
Take updates from https://github.com/dotnet/runtime/issues/34978#issuecomment-614845405.

Reply:
Huh ... I changed this in my machine. Must not have saved before submitting.